dm/pkg/shardddl/optimism/lock.go

```diff
@@ -248,7 +248,7 @@ func (l *Lock) TrySync(info Info, tts []TargetTable) (newDDLs []string, cols []s
 			if col, err2 := GetColumnName(l.ID, ddls[idx], ast.AlterTableAddColumns); err2 != nil {
 				return newDDLs, cols, err2
 			} else if len(col) > 0 && l.IsDroppedColumn(info.Source, info.UpSchema, info.UpTable, col) {
-				return newDDLs, cols, terror.ErrShardDDLOptimismTrySyncFail.Generate(
+				return newDDLs, cols, terror.ErrShardDDLOptimismAddNotFullyDroppedColumn.Generate(
 					l.ID, fmt.Sprintf("add column %s that wasn't fully dropped in downstream. ddl: %s", col, ddls[idx]))
```

Comment:
How about move the error message: 'add column that wasn't ...' to terror.ErrShardDDLOptimismAddNotFullyDroppedColumn, so we only need terror.ErrShardDDLOptimismAddNotFullyDroppedColumn.Generate(col,ddl)

Reply:
We have several kinds of reasons and error messages for `ErrShardDDLOptimismAddNotFullyDroppedColumn`. We may still need to generate message in this way.